peekingduck/pipeline/nodes/dabble/statisticsv1/utils.py

```diff
@@ -0,0 +1,227 @@
+            operand_raw = self.expr[op_idx_end:]
+            self.condition["operand"] = _get_operand(operand_raw)  # type: ignore
+        else:
+            target_attr = self.expr
+
+        self.data_type, self.keys = _get_data_type_and_keys(target_attr)
+
+
+def _get_method_expr(all_methods: Dict[str, Union[str, None]]) -> Tuple[str, str]:
+    """Gets the only non-null key value pair from a dictionary."""
+    num_methods = 0
+    for method, expr in all_methods.items():
+        if expr:
+            num_methods += 1
+            selected_method, selected_expr = method, expr
+
+    if num_methods < 1:
+        raise ValueError(
+            "No methods selected in config, but one method needs to be selected to proceed."
+        )
+    if num_methods > 1:
+        raise ValueError(
+            f"{num_methods} methods selected in config, but only one method should be selected "
+            f"to proceed."
+        )
+    return selected_method, selected_expr
+
+
+def _get_data_type_and_keys(target_attr: str) -> Tuple[str, List[str]]:
+    """Extract the name of data_type and its associated keys in a list, if any."""
+    target_attr = re.sub(r"'|\"| ", "", target_attr)
+    keys = re.findall(r"\[(.*?)\]", target_attr)
+    if not keys:
+        data_type = target_attr
+        keys = []
+    else:
+        data_type = target_attr.split("[")[0]
+
+    return data_type, keys
+
+
+def _get_operand(operand_raw: str) -> Union[str, float]:
+    """Removes leading and trailing spaces, and return as a string or float."""
+    operand_raw = operand_raw.strip()
+    is_string = re.findall("'|\"", operand_raw)
+    if is_string:
+        operand = re.sub("'|\"", "", operand_raw)
+        return operand
+    try:
+        return float(operand_raw)
+    except ValueError as error:
+        raise ValueError(
+            f"The detected operand here is: {operand_raw}."
+            f"If the operand is intended to be a string, ensure that it is enclosed by single "
+            f"or double quotes. Else, it is assumed to be of integer or float type, and will be "
+            f"subsequently converted into a float."
+        ) from error
+
+
+def _deep_get_value(
+    data: Dict[str, Any], keys: List[str]
+) -> Union[int, float, list, dict]:
+    """Recursively goes through the keys of a dictionary to obtain the final value."""
+    if not keys:
+        return data
+    key = keys.pop(0)
+    return _deep_get_value(data[key], keys)
+
+
+def _apply_method(
+    target_attr: Any,
+    method: str,
+    condition: Dict[str, Any],
+) -> Union[int, float, None]:
+    """Applies a method and optional condition to a target attribute."""
+    if not target_attr:
+        return None
+    allowable_types = {
+        "identity": (int, float),
+        "length": (dict, list),
+        "minimum": (dict, list),
+        "maximum": (dict, list),
+        "cond_count": (list),
+    }
+    _check_type(target_attr, method, allowable_types[method])  # type: ignore
```

Comment:
There are 2 reasons why I placed this in the `run` stage instead of `init`:
1. `target_attr` is only accessible during `run` when inputs are given to this Node. Of course, one way is to do this check for the 1st input and ignore for subsequent inputs...
2. ... which leads to another consideration, where users may be using their own custom data types from their custom nodes. If the data type changes due to a mistake (e.g. `int` most of the time, but `str` in some edge cases), and the check is only done for the 1st input, when it becomes a `str` it will fail without a PeekingDuck raised error.